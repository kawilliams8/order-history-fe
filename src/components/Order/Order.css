.Order {
  background-color: white;
  box-shadow: 10px 10px 5px grey;
  margin: 40px;
}