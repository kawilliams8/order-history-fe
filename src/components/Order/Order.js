import React from 'react';
import '../Order/Order.css';

const Order = ({ data }) => {
  const { id, img, name, description, price } = data;
  return (
    <article className="Order">
      <img className="productImage" src={img} alt={description}/>
      <h3 className="productName" >{name}</h3>
      <h3 className="productDescription" >{description}</h3>
      <h3 className="productPrice" >${price}</h3>
      <button></button>
    </article>
  )
}

export default Order;