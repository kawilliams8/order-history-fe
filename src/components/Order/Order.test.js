import React from 'react';
import { shallow } from 'enzyme';
import Order from '../Order/Order';

describe('Order', () => {
3
  let mockOrder;
  beforeEach(() => {
    mockOrder = {
      id: 1,
      img: "image.png",
      name: "Mock Order",
      description: "This is a mock description",
      price: 3
    }
  })

  it('should match the snapshot', () => {
    const removeOrderMock = jest.fn();
    const wrapper = shallow(<Order data={mockOrder} key={1} removeOrder={removeOrderMock} />)
    expect(wrapper).toMatchSnapshot()
  });

  it('should call the removeOrder method and pass the id', () => {
    const removeOrderMock = jest.fn();
    const wrapper = shallow(<Order data={mockOrder} key={1} removeOrder={removeOrderMock}/>)
    wrapper.find('button').simulate('click')
    expect(removeOrderMock).toHaveBeenCalled();
    expect(removeOrderMock).toHaveBeenCalledWith(1);
  })

})