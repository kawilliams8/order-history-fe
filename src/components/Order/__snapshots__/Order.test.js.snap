// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match the snapshot 1`] = `
<article
  className="Order"
>
  <img
    alt="This is a mock description"
    className="productImage"
    src="image.png"
  />
  <h3
    className="productName"
  >
    Mock Order
  </h3>
  <h3
    className="productDescription"
  >
    This is a mock description
  </h3>
  <h3
    className="productPrice"
  >
    $
    3
  </h3>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    Remove From History
  </button>
</article>
`;
